src/accountsdb/db.zig

```diff
@@ -2576,32 +2589,28 @@ test "purge accounts in cache works" {
     const n_accounts = 3;
 
     // we dont defer deinit to make sure that they are cleared on purge
-    var pubkeys = try allocator.alloc(Pubkey, n_accounts);
-    var accounts = try allocator.alloc(Account, n_accounts);
+    var pubkeys: [n_accounts]Pubkey = undefined;
+    var accounts: [n_accounts]Account = undefined;
 
-    for (0..n_accounts) |i| {
-        pubkeys[i] = Pubkey.random(rng);
-        accounts[i] = try Account.random(allocator, rng, i % 1_000);
+    for (&pubkeys, &accounts, 0..) |*pubkey, *account, i| {
+        errdefer for (accounts[0..i]) |prev_account| prev_account.deinit(allocator);
+        pubkey.* = Pubkey.random(rng);
+        account.* = try Account.random(allocator, rng, i % 1_000);
     }
+    defer for (accounts) |account| account.deinit(allocator);
```

Comment:
Ah forgot to remove the comment. A part of this changeset involved improving the memory safety of `putAccountSlice`, which was causing leaks during fuzz testing, which involved making it clone the slices it's passed & the accounts in the slice, which caused these to no longer need to live beyond their scope.

Note: This isn't an entirely ideal situation, since it would be good to still have the option to pre-allocate these before adding them to accountsdb, but that should receive some special attention and care to implement it in such a way that it is safe and has obvious semantics.

Reply:
9b83c2f6f4afc4c0ba339f1924ff021e977828fe